### PR TITLE
VS Insertion pipelines fix

### PR DESCRIPTION
**Pull request by JanProvaznik**
The new VS insertion pipelines introduced in https://github.com/dotnet/msbuild/pull/10920 did not work as expected, there was a mismatch which Artifacts were used. This PR addresses that and adds parameters to enable insertion from any branch. 
Mitigate Guardian warning by deleting intermediate sourcebuild package.

### Testing
Manual pipeline runs.

### CODE REVIEW

**Comment by JanProvaznik** [comment]
Merging this should wait for validating the insertions are valid after current issues get resolved.

File: azure-pipelines/vs-insertion-experimental.yml [hunk 1]
```diff
@@ -28,25 +20,25 @@ extends:
...
+        InsertJsonValues: Microsoft.Build.vsman{$(MSBuild_ExtApisPackageVersion)+$(resources.pipeline.MSBuild.sourceVersion)}=https://vsdrop.corp.microsoft.com/file/v1/Products/DevDiv/DotNet-msbuild-Trusted/$(resources.pipeline.MSBuild.sourceBranch)/$(resources.pipeline.MSBuild.runName);Microsoft.Build.vsman,Microsoft.Build.Arm64.vsman{$(MSBuild_ExtApisPackageVersion)+$(resources.pipeline.MSBuild.sourceVersion)}=https://vsdrop.corp.microsoft.com/file/v1/Products/DevDiv/DotNet-msbuild-Trusted/$(resources.pipeline.MSBuild.sourceBranch)/$(resources.pipeline.MSBuild.runName);Microsoft.Build.Arm64.vsman,Microsoft.Build.UnGAC.vsman=https://vsdrop.corp.microsoft.com/file/v1/Products/DevDiv/DotNet-msbuild-Trusted/$(resources.pipeline.MSBuild.sourceBranch)/$(resources.pipeline.MSBuild.runName);Microsoft.Build.UnGAC.vsman
+        InsertPayloadName: MSBuild $(resources.pipeline.MSBuild.sourceBranch) $(MSBuild_ExtApisPackageVersion)
+        TeamName: msbuild
+        TeamEmail: msbuild@microsoft.com
```
Comment (YuliiaKovalova) [inline azure-pipelines/vs-insertion-experimental.yml:34]: please use this email
msbtm@microsoft.com

File: azure-pipelines/vs-insertion.yml [hunk 2]
```diff
@@ -150,6 +154,14 @@ extends:
             artifactName: 'PackageArtifacts'
             targetPath: '$(Build.ArtifactStagingDirectory)/PackageArtifacts'
         steps:
+        - task: Powershell@2
```
Comment (YuliiaKovalova) [inline azure-pipelines/vs-insertion.yml:157]: please add comment why we do that

File: azure-pipelines/vs-insertion.yml [hunk 3]
```diff
@@ -106,41 +125,26 @@ extends:
...
-          SymbolsEmailContacts: raines
-          SymbolsFeatureName: MSBuild
-          SymbolsSymwebProject: DDE
-          SymbolsUncPath: '\\cpvsbuild\drops\MSBuild\symbols\$(Build.DefinitionName)\$(Build.BuildNumber)\symbols.archive'
+          InsertDescription: Insert MSBuild $(MSBuild_CI_BuildNumber) from the branch $(SourceBranchName) at commit $(MSBuild_CI_SourceVersion). Corresponding package version is $(MSBuild_ExtApisPackageVersion)
+          InsertJsonValues: Microsoft.Build.vsman{$(MSBuild_ExtApisPackageVersion)+$(MSBuild_CI_SourceVersion)}=https://vsdrop.corp.microsoft.com/file/v1/Products/DevDiv/DotNet-msbuild-Trusted/$(SourceBranchName)/$(MSBuild_CI_BuildNumber);Microsoft.Build.vsman,Microsoft.Build.Arm64.vsman{$(MSBuild_ExtApisPackageVersion)+$(MSBuild_CI_SourceVersion)}=https://vsdrop.corp.microsoft.com/file/v1/Products/DevDiv/DotNet-msbuild-Trusted/$(SourceBranchName)/$(MSBuild_CI_BuildNumber);Microsoft.Build.Arm64.vsman,Microsoft.Build.UnGAC.vsman=https://vsdrop.corp.microsoft.com/file/v1/Products/DevDiv/DotNet-msbuild-Trusted/$(SourceBranchName)/$(MSBuild_CI_BuildNumber);Microsoft.Build.UnGAC.vsman
```
Comment (YuliiaKovalova) [inline azure-pipelines/vs-insertion.yml:139]: is it possible to move https://vsdrop.corp.microsoft.com/file/v1/Products/DevDiv/DotNet-msbuild-Trusted to a variable and use it for the packages?

**Review by YuliiaKovalova — APPROVED** [review]

File: azure-pipelines/vs-insertion-experimental.yml [hunk 4]
```diff
@@ -1,13 +1,7 @@
...
-  InsertCustomScriptExecutionCommand: $(Pipeline.Workspace)\xsd\Update-MSBuildXsds.ps1
-  InsertDescription: Insert MSBuild $(Build.BuildNumber) from the branch $(Build.SourceBranchName) at commit $(Build.SourceVersion). Corresponding package version is $(MSBuild_ExtApisPackageVersion)
-  InsertJsonValues: Microsoft.Build.vsman{$(MSBuild_ExtApisPackageVersion)+$(Build.SourceVersion)}=https://vsdrop.corp.microsoft.com/file/v1/Products/DevDiv/DotNet-msbuild-Trusted/$(Build.SourceBranchName)/$(Build.BuildNumber);Microsoft.Build.vsman,Microsoft.Build.Arm64.vsman{$(MSBuild_ExtApisPackageVersion)+$(Build.SourceVersion)}=https://vsdrop.corp.microsoft.com/file/v1/Products/DevDiv/DotNet-msbuild-Trusted/$(Build.SourceBranchName)/$(Build.BuildNumber);Microsoft.Build.Arm64.vsman,Microsoft.Build.UnGAC.vsman=https://vsdrop.corp.microsoft.com/file/v1/Products/DevDiv/DotNet-msbuild-Trusted/$(Build.SourceBranchName)/$(Build.Build.Number);Microsoft.Build.UnGAC.vsman
-  InsertPayloadName: MSBuild $(Build.SourceBranchName) $(Build.BuildNumber)
-  TeamName: msbuild
-  TeamEmail: msbuild@microsoft.com
+name: $(Date:yyyyMMdd).$(Rev:r)
```
[the remark this answers is not in the document]
Comment (JanProvaznik) [reply azure-pipelines/vs-insertion-experimental.yml:3]: This is not the title of the PR but the name of the run in pipeline view.

[the remark this answers is not in the document]
Comment (maridematte) [reply azure-pipelines/vs-insertion-experimental.yml:3]: Ooohh, I see. I thought it was the name of the insertion PR created. Thank you for clarifying.

File: azure-pipelines/vs-insertion-experimental.yml [hunk 5]
```diff
@@ -56,8 +71,29 @@ extends:
...
         steps:
+        # the CI build creates a sourcebuild intermediate package that is not signed, remove it to avoid warning from Guardian
+        - task: Powershell@2
+          name: RemoveSourceBuildPackage
+          displayName: Remove sourcebuild package
+          inputs:
+            targetType: inline
+            script: |
+              $folder = "$(Build.ArtifactStagingDirectory)/PackageArtifacts"
+              Get-ChildItem -Path $folder -Filter 'Microsoft.SourceBuild.Intermediate*.nupkg' | Remove-Item -Force
```
Comment (rainersigwald) [inline azure-pipelines/vs-insertion-experimental.yml:82]: Can you add an exclude pattern to the `PackageArtifacts` download? Something like `!**/Microsoft.SourceBuild.Intermediate*.nupkg`?

**Review by surayya-MS — APPROVED** [review]

**Pull request merged**